action.yaml

```diff
@@ -99,8 +100,11 @@ runs:
         lxc network attach lxdbrstack devstack eth0 eth0 --project devstack
         lxc config device set devstack eth0 ipv4.address ${{ steps.parse-network.outputs.devstack-ip }} --project devstack
         lxc start devstack --project devstack
+        echo waiting for initialization
         timeout 120s bash -c 'until lxc exec devstack --project devstack -- bash -c "[ -d /home/ubuntu ]"; do sleep 1 && echo .; done; echo'
-        lxc exec devstack --user 1000 --cwd /home/ubuntu --project devstack -- git clone -b stable/yoga https://opendev.org/openstack/devstack
+        echo waiting for network
+        timeout 120s bash -c 'until lxc exec devstack --project devstack -- bash -c "curl -sSL https://github.com/openstack/devstack.git -m 3 -o /dev/null"; do sleep 1; done; echo'
```

Comment:
`curl` already have retry options
See
https://curl.se/docs/manpage.html#--retry
https://curl.se/docs/manpage.html#--retry-delay
https://curl.se/docs/manpage.html#--retry-max-time

I'd rather use them

Reply:
Good idea! Updated.

Reply:
The `curl` version inside the LXD container is too old and it doesn't have the `--retry-all-errors` option. Without that network failure like DNS error or connection refused will not be retried. I will revert to the previous commit.